core/src/main/scala/magnolia1/impl.scala

```diff
@@ -96,6 +96,39 @@ object CaseClassDerivation:
       } { params => product.fromProduct(Tuple.fromArray(params)) }
     }
 
+  inline def paramsFromMapsStep[Typeclass[_], A, l, p](
+      annotations: Map[String, List[Any]],
+      inheritedAnnotations: Map[String, List[Any]],
+      typeAnnotations: Map[String, List[Any]],
+      repeated: Map[String, Boolean],
+      defaults: Map[String, Option[() => Any]],
+      idx: Int
+  ): CaseClass.Param[Typeclass, A] =
+    val label = constValue[l].asInstanceOf[String]
+    val tc = new SerializableFunction0[Typeclass[p]]:
+      override def apply(): Typeclass[p] = summonInline[Typeclass[p]]
+    val d =
+      defaults.get(label).flatten match {
+        case Some(evaluator) =>
+          new SerializableFunction0[Option[p]]:
+            override def apply(): Option[p] =
+              val v = evaluator()
+              if ((v: @unchecked).isInstanceOf[p]) new Some(v).asInstanceOf[Option[p]]
+              else None
+        case _ =>
+          returningNone.asInstanceOf[SerializableFunction0[Option[p]]]
+      }
+    paramFromMaps[Typeclass, A, p](
+      label,
+      CallByNeed.createLazy(tc),
+      CallByNeed.createValueEvaluator(d),
+      repeated,
+      annotations,
+      inheritedAnnotations,
+      typeAnnotations,
+      idx
+    )
+
   inline def paramsFromMaps[Typeclass[_], A, Labels <: Tuple, Params <: Tuple](
```

Comment:
It would be useful to add a comment explaining why the function uses this unrolled recursion.